R/plots.R

```diff
@@ -535,3 +544,190 @@ cumulative_value.plot <- function(data,
+    plot(lifted, notes = paste(
+      # "Locations:", Market$location,
+      # "\n Rank:", Market$rank,
+      # "\n Treatment Periods:", Market$duration,
+      # "\n Effect Size: ", Market$EffectSize
+    )),
+    PowerPlot_graph,
+    ncol = 1,
+    nrow = 2,
+    bottom = paste(
+      "Rank:", Market$rank,
+      "\n Locations:", Market$location,
+      "\n Treatment Periods:", Market$duration,
+      "\n Effect Size: ", Market$EffectSize
+    )))
+
+}
+
+
+#' Plotting function for historical correlations between the test market's
+#' and total markets KPI.
+#'
+#' @description
+#'
+#' `plotCorrels` takes a data frame used for GeoLift to generate
+#' a plot that show historical similarities in KPI levels between the
+#' test markets and the aggregation of all locations (total).
+#'
+#' @param data A data.frame containing the historical conversions by
+#' geographic unit. It requires a "locations" column with the geo name,
+#' a "Y" column with the outcome data (units), a time column with the indicator
+#' of the time period (starting at 1), and covariates.
+#' @param locs List of markets to use in the calculation of the correlations.
+#' @param scaled A logic flag indicating whether to plot the scaled values of
+#' the KPI metric (standardized by the largest historical value). Set to FALSE
+#' by default to plot the observed KPI levels.
+#' @param KPI_id Outcome variable.
+#' @param dtw Emphasis on Dynamic Time Warping (DTW), dtw = 1 focuses exclusively
+#' on this metric while dtw = 0 (default) relies on correlations only.
+#'
+#' @return
+#' A plot of the historical values of the test market and the aggregation of
+#' all control markets.
+#'
+#' @export
+plotCorrels <- function(data,
+                        locs = c(),
+                        scaled = TRUE,
+                        KPI_id = "",
+                        dtw = 0){
+
+  if(!(all(tolower(locs) %in% tolower(unique(data$location))))){
+    warning("Please specify a valid set of test locations.")
+    return(NULL)
+  }
+
+  data_aux <- AppendAgg(data, locs = locs)
+  correl <- GetCorrel(data, locs = locs, dtw = dtw)
+
+  if(scaled == TRUE){
+    data_aux$Yscaled <- 0
+    data_aux$Yscaled[data_aux$location == "control_markets"] <- data_aux$Y[data_aux$location == "control_markets"] / max(data_aux[data_aux$location == "control_markets",]$Y)
+    data_aux$Yscaled[data_aux$location == "test_markets"] <- data_aux$Y[data_aux$location == "test_markets"] / max(data_aux[data_aux$location == "test_markets",]$Y)
+
+    GeoPlot(data_aux,
+            Y_id = "Yscaled",
+            time_id = "time",
+            location_id = "location",
+            KPI_id = KPI_id,
+            notes = paste0("Correlation: ", round(correl, 4)))
+  } else{
+
+    GeoPlot(data_aux,
+            Y_id = "Y",
+            time_id = "time",
+            location_id = "location",
+            KPI_id = KPI_id,
+            notes = paste0("Correlation: ", round(correl, 4)))
```

Comment:
Maybe this can be unified? I am thinking something like:
```
if (scaled == TRUE) {
....run transformations to Y....
Y_id <- "Yscaled"
} else {
Y_id <- "Y"
}
GeoPlot(..., Y_id=Y_id, ...)

Reply:
Good idea, just implemented something along those lines.